### DISCUSSION TITLE
V11 Beta Support

### DISCUSSION THREAD

**Discussion by balloman** [rnmapbox/maps#3039] [Q&A] [answered]
I'm looking to start an application and a big win we need is v11 support with the new Mapbox Standard style. How do we get this to work with the library?

**Comment by mfazekas** [comment]
V11 api requires some changes.  I've started a branch for it but far from finished. Ideally it'll be a conditional compilation, for v11 using a feature flag. And once v11 stable we'll switch over completely.

https://github.com/rnmapbox/maps/pull/3040

**Reply by mfazekas** [reply]
initial v11 support is on main now. Android is quite complete iOS needs some more work.

To use it on iOS, add the following to your Podfile:

```ruby
$RNMapboxMapsUseV11 = true # use 11 version
$RNMapboxMapsVersion = '= 11.0.0-beta.3'
```

To use it on android add the following to your top-level `build.gradle`  into `buildscript` > `ext`
```gradle
RNMapboxMapsUseV11 = true
RNMapboxMapsVersion = '11.0.0-beta.3'
```

**Reply by mfazekas** [reply]
I'm also thinking about adding support for styeImport configs like this:

```jsx
<MapView>
   <StyleImport 
      id="basemap" existing
      config={{lightPreset: "dusk"}}
   />
</MapView>
```

**Reply by balloman** [reply]
Hmm, I've tried adding the required items to my podfile but cannot get past this build error @mfazekas :
```
[RUN_FASTLANE] 
❌  (/var/folders/9l/cgk75l2d0716q7mvx6bszzsw0000gn/T/eas-build-local-nodejs/c87e385a-21fa-48f9-91a2-cf0dfa166605/build/social-app/node_modules/@rnmapbox/maps/ios/RCTMGL-v10/MGLModule.swift:3:8)

  1 | import Foundation
  2 | import MapboxMaps
> 3 | import MapboxMobileEvents
    |        ^ no such module 'MapboxMobileEvents'
  4 | 
  5 | 
  6 | let DEFAULT_SOURCE_ID = "composite";

[RUN_FASTLANE] 
❌  (/var/folders/9l/cgk75l2d0716q7mvx6bszzsw0000gn/T/eas-build-local-nodejs/c87e385a-21fa-48f9-91a2-cf0dfa166605/build/social-app/node_modules/@rnmapbox/maps/ios/RCTMGL-v10/MGLModule.swift:3:8)

  1 | import Foundation
  2 | import MapboxMaps
> 3 | import MapboxMobileEvents
    |        ^ no such module 'MapboxMobileEvents'
  4 | 
  5 | 
  6 | let DEFAULT_SOURCE_ID = "composite";
```

**Reply by balloman** [reply]
I think I fixed it by pulling the version from github, but unfortunately, now I am presented with this error: 
```
[RUN_FASTLANE] 
❌  (/var/folders/9l/cgk75l2d0716q7mvx6bszzsw0000gn/T/eas-build-local-nodejs/64eb8e50-494b-49b4-81be-48c05886205e/build/social-app/node_modules/@rnmapbox/maps/ios/RNMBX/rnmapbox_maps-Swift.h.pre:9:9)

  7 | @end
  8 | 
> 9 | #import <rnmapbox_maps-Swift.h>
    |         ^ 'rnmapbox_maps-Swift.h' file not found
```

If it's possible to provide a tutorial for how to enable V11, that would be great!

**Reply by mfazekas** [reply]
You should be using beta released releases of 10.1.0. Like `[10.1.0-beta.9](https://www.npmjs.com/package/@rnmapbox/maps/v/10.1.0-beta.9)` 

Are you using expo? Which version?

**Reply by balloman** [reply]
Hi, I'm using this version: https://www.npmjs.com/package/@rnmapbox/maps/v/10.1.0-beta.9
Additionally, I'm using Expo bare workflow, and I've tried building using XCode as well, but I get the same error. I think the issue is that there is no rn-mapbox_maps-Swift.h file in the ios folder that I can see? I'm not completely sure how it works.
My podfile looks as so:
```ruby
$RNMapboxMapsImpl = 'mapbox'
$RNMapboxMapsDownloadToken = ''
$RNMapboxMapsVersion = '= 11.0.0-beta.4' (I've also tried without the equal sign)
$RNMapboxMapsUseV11 = true
```

**Reply by mfazekas** [reply]
This header file should be generated by Xcode, to [export stuff in Swift to objective-C](https://developer.apple.com/documentation/swift/importing-swift-into-objective-c). 

Can you try to reproduce from an empty expo project.
Also what is your Xcode version?

**Reply by balloman** [reply]
I've nailed down the issue to ios static frameworks, which I have enabled in my project. When disabled, it seems to build fine, but when disabled, I encounter the error. I'll try to look into how I can resolve this. I'm trying to use react-native-firebase which requires use_frameworks! static so maybe the two libraries are just incompatible

**Reply by balloman** [reply]
@mfazekas Do you have any insight on how this file gets generated? I'm working on seeing what work would need to be done to get this library to support use_frameworks but I I'm still new to Swift

**Reply by balloman** [reply, marked as answer]
I was able to fix it by changing the rnmapbox_maps-Swift.h.pre file to import <rnmapbox_maps/rnmapbox_maps-Swift.h.pre> instead and that fixed it for me

**Reply by balloman** [reply]
This might need to be mentioned as a workaround for users that are using static frameworks @mfazekas

**Reply by mfazekas** [reply]
nice, can you add a PR, I'll try that, it might now work without use_frameowkrs! but then we can add some conditional to support that

**Reply by balloman** [reply]
Yeah, in my testing it does NOT work if you're not using static frameworks

**Reply by mfazekas** [reply]
I've just committed #3102

You should be able to use with frameworks, by either setting `USE_FRAMEWORKS` shell environment variable. Or setting the `$RNMapboxMapsUseFrameworks` ruby global

**Reply by balloman** [reply]
I think that whatever you added in your commit works even without having to set the USE_FRAMEWORKS variable. Maybe it was something up with the file extension. Thanks! I'll try to add a PR to make this work with expo managed
